spring-vault-core/src/main/java/org/springframework/vault/core/lease/SecretLeaseContainer.java

```diff
@@ -592,11 +593,16 @@ protected Lease doRenewLease(RequestedSecret requestedSecret, Lease lease) {
 
 	@SuppressWarnings("unchecked")
 	private Lease renew(Lease lease) {
+		final Map<String, String> leaseRenewalData = new HashMap<>();
```

Comment:
Typically, we avoid `final` in method bodies since our upgrade to Java 1.8. I'll take care of this during the merge.

RE your comment on immutable maps: We also do not require immutable maps as the map does not leave this method.

Reply:
Anything else you want me to take care of or is this ok?